### PR TITLE
Support for transfer manager to download only part of file

### PR DESCRIPTION
**Pull request by rahul003**
*Description of changes:*
This feature arises from our use of the SDK in the open source project Tensorflow. We would like to use transfer manager to download a file specified by an offset and given number of bytes. 

I thought this feature makes less sense when downloading to a file, hence I only added support when downloading to a stream.

*Check all that applies:*
- [x] Did a review by yourself.
- [x] Added proper tests to cover this PR. (If tests are not applicable, explain.)
- [x] Checked if this PR is a breaking (APIs have been changed) change.
- [x] Checked if this PR will _not_ introduce cross-platform inconsistent behavior.
- [x] Checked if this PR would require a ReadMe/Wiki update.

Check which platforms you have built SDK on to verify the correctness of this PR.
- [x] Linux
- [ ] Windows
- [ ] Android
- [ ] MacOS
- [ ] IOS
- [ ] Other Platforms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by rahul003** [comment]
I don't have access to the other platforms listed above. Can we use CI to test those?

**Comment by rahul003** [comment]
@singku @marcomagdy Could you please help review this? This feature helps us use transfer manager in TensorFlow and improve the read performance by an order of magnitude (~12x). I'm also in AWS. If you want to discuss something, I'm available to talk over Chime too.

File: aws-cpp-sdk-transfer/include/aws/transfer/TransferHandle.h [hunk 1]
```diff
@@ -154,6 +154,14 @@ namespace Aws
              */
             TransferHandle(const Aws::String& bucketName, const Aws::String& keyName, CreateDownloadStreamCallback createDownloadStreamFn, const Aws::String& targetFilePath = "");
 
+            /**
+             * Alternate DOWNLOAD constructor
+             */
+            TransferHandle(const Aws::String& bucketName, const Aws::String& keyName,
```
Comment (singku-china) [inline aws-cpp-sdk-transfer/include/aws/transfer/TransferHandle.h:160]: `offset` here is not clear of what kind of `offset`, I'd use `fileOffset` and `downloadBytes`. Also document that `fileOffset` currently only used for download.

File: aws-cpp-sdk-transfer/include/aws/transfer/TransferHandle.h [hunk 2]
```diff
@@ -254,6 +262,10 @@ namespace Aws
             */
             void UpdateBytesTransferred(uint64_t amount) { m_bytesTransferred.fetch_add(amount); }
 
+            /**
+             * The offset from which to start downloading
+             */
+            inline uint64_t GetBytesOffset() const { return m_offset; }
```
Comment (rahul003) [inline aws-cpp-sdk-transfer/include/aws/transfer/TransferHandle.h:268]: I chose to keep t

**Comment by rahul003** [comment]
@singku-china Thanks for your comments, I addressed them. Please review again.

**Comment by rahul003** [comment]
Hi @singku-china , how did that run go?

**Comment by KaibaLopez** [comment]
Hi @rahul003 ,
this PR was merged on version 1.7.336 so I'll be closing it.
Thank you for your contribution!

**Pull request closed without merging**